libwasmvm/src/test_utils.rs

```diff
@@ -0,0 +1,103 @@
+#![cfg(test)]
+
+use cosmwasm_std::{Decimal, Uint128};
+use std::str::FromStr as _;
+
+/// Asserts that two expressions are approximately equal to each other.
+///
+/// The `max_rel_diff` argument defines the maximum relative difference
+/// of the `left` and `right` values.
+///
+/// On panic, this macro will print the values of the arguments and
+/// the actual relative difference.
+///
+/// Like [`assert_eq!`], this macro has a second form, where a custom
+/// panic message can be provided.
+#[macro_export]
+macro_rules! assert_approx_eq {
+    ($left:expr, $right:expr, $max_rel_diff:expr $(,)?) => {{
+        $crate::test_utils::assert_approx_eq_impl($left, $right, $max_rel_diff, None);
+    }};
+    ($left:expr, $right:expr, $max_rel_diff:expr, $($args:tt)+) => {{
+        $crate::test_utils::assert_approx_eq_impl($left, $right, $max_rel_diff, Some(format!($($args)*)));
+    }};
+}
+
+#[track_caller]
+fn assert_approx_eq_impl<U: Into<Uint128>>(
+    left: U,
+    right: U,
+    max_rel_diff: &str,
+    panic_msg: Option<String>,
+) {
+    let left = left.into();
+    let right = right.into();
+    let max_rel_diff = Decimal::from_str(max_rel_diff).unwrap();
+
+    let largest = std::cmp::max(left, right);
+    let smallest = std::cmp::min(left, right);
+    let rel_diff = Decimal::from_ratio(largest - smallest, largest);
```

Comment:
Isn't that the same as this?

```suggestion
    let rel_diff = Decimal::from_ratio(left.abs_diff(right), largest);
```

The suggestion would be coser to to formula from Wikipedia

Reply:
Where does `abs_diff` come from? These types don't have that method that I've seen

Reply:
From https://github.com/CosmWasm/cosmwasm/pull/1334